### PR TITLE
Clarify init container concept

### PR DESCRIPTION
**Pull request by vishnuswmech**
I have made the Init container concepts from both Pod and Deployment perspective.

and This is my first contribution and let me know if any changes needs to be done on PR.

Issue

Fixes: https://github.com/kubernetes/website/issues/38271

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
Thank you for the PR.

Here's some feedback
- avoid implying that manifests have to be YAML; other formats can work too (notably JSON)
- [Use the active voice](https://kubernetes.io/docs/contribute/style/style-guide/#use-active-voice)
- you don't need to explain too much about how Deployments work; signposting to https://kubernetes.io/docs/concepts/workloads/controllers/ may be enough
  (bear in mind that Deployment is only one of several workload management APIs in Kubernetes)

We shouldn't merge this as-is, but I hope we can help you tweak it so it's good to go in.

**Comment by vishnuswmech** [comment]
Thanks @sftim for the suggestions.

Sure, Will work on it.

**Comment by vishnuswmech** [comment]
@sftim I have made the changes based on your suggestions. Kindly review this and let me know if any further corrections needed.

File: content/en/docs/concepts/workloads/pods/init-containers.md [hunk 1]
```diff
@@ -291,10 +291,9 @@ is in the `Pending` state but should have a condition `Initialized` set to false
 If the Pod [restarts](#pod-restart-reasons), or is restarted, all init containers
 must execute again.
 
-Changes to the init container spec are limited to the container image field.
-Altering an init container image field is equivalent to restarting the Pod.
+Changes to the init container spec are limited to the container image field. Altering the image field of the init container does not restart or recreate the Pod.
 
-Because init containers can be restarted, retried, or re-executed, init container
+However, since init containers can be restarted, retried, or re-executed, init container
 code should be idempotent. In particular, code that writes to files on `EmptyDirs`
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/init-containers.md:297]: (optionally)
```suggestion
code should be idempotent. In particular, code that writes into any `emptyDir` volume
```

**Review by sftim — COMMENTED** [review]
/lgtm

I think there's room to improve, but I want to be clear: further tweaks are optional.

**Comment by reylejano** [comment]
@vishnuswmech , what do you think of sftim's [suggestions](https://github.com/kubernetes/website/pull/47946#discussion_r1763581830)

**Comment by vishnuswmech** [comment]
Thanks @sftim , now got some idea.

@reylejano I will make the necessary changes and will commit it shortly.

File: content/en/docs/concepts/workloads/pods/init-containers.md [hunk 2]
```diff
@@ -291,10 +291,9 @@ is in the `Pending` state but should have a condition `Initialized` set to false
 If the Pod [restarts](#pod-restart-reasons), or is restarted, all init containers
 must execute again.
 
-Changes to the init container spec are limited to the container image field.
-Altering an init container image field is equivalent to restarting the Pod.
+Changes to the init container spec are limited to the container image field. Altering the image field of the init container does not restart or recreate the Pod.
 
-Because init containers can be restarted, retried, or re-executed, init container
+However, since init containers can be restarted, retried, or re-executed, init container
```
[the remark this answers is not in the document]
Comment (vishnuswmech) [reply content/en/docs/concepts/workloads/pods/init-containers.md]: Done

File: content/en/docs/concepts/workloads/pods/init-containers.md (hunk 1 above)
Comment (vishnuswmech) [reply content/en/docs/concepts/workloads/pods/init-containers.md]: Make sense. It is done.

File: content/en/docs/concepts/workloads/pods/init-containers.md [hunk 3]
```diff
@@ -291,10 +291,9 @@ is in the `Pending` state but should have a condition `Initialized` set to false
 If the Pod [restarts](#pod-restart-reasons), or is restarted, all init containers
 must execute again.
 
-Changes to the init container spec are limited to the container image field.
-Altering an init container image field is equivalent to restarting the Pod.
+Changes to the init container spec are limited to the container image field. Altering the image field of the init container does not restart or recreate the Pod.
```
[the remark this answers is not in the document]
Comment (vishnuswmech) [reply content/en/docs/concepts/workloads/pods/init-containers.md]: It is done.

**Comment by vishnuswmech** [comment]
@sftim  @reylejano  Made the required changes based on your suggestions. Kindly check now and let me know if any further modifications needed.

**Comment by vishnuswmech** [comment]
@sftim Kindly approve this.

**Comment by T-Lakshmi** [comment]
All suggestions are seems incorporated and preview looks good to me.
[Preview page](https://deploy-preview-47946--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/pods/init-containers/#detailed-behavior)
/lgtm

**Comment by vishnuswmech** [comment]
Thanks @T-Lakshmi for your review.

**Comment by vishnuswmech** [comment]
@sftim Kindly approve this PR.

**Review by sftim — COMMENTED** [review]
Thanks

/approve

**Pull request merged**